apis/cluster/v1/commons.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ClusterState string
+
+const (
+	ClusterStateJoin  ClusterState = "Join"
+	ClusterStateLeave ClusterState = "Leave"
+)
+
+// ResourceUsage contains the observed resource usage of a member cluster.
+type ResourceUsage struct {
```

Comment:
some of the fields here are also in preview

Reply:
Per earlier discussions, a beta-level hint comment is added.